src/logger.py

```
@@ -27,27 +27,30 @@ def __init__(self, filename: str = "logs/llm_requests.json"):
-    def log(self, payload: dict[str, Any]) -> None:
+    def log(self, payload: Mapping[str, object]) -> None:
         """
         Sanitizes and logs the request payload as a pretty-printed JSON object.
         """
         try:
             # Create a shallow copy
-            log_entry = payload.copy()
+            log_entry: dict[str, object] = dict(payload)
 
             # Inject timestamp for context
             log_entry["_timestamp"] = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
```

Ruff (0.14.10)

`datetime.datetime.now()` called without a `tz` argument

(DTZ005)